Remove wpcom_vip_get_page_by_path

BrookeDot opened the pull request:
Removes the now deprecated `wpcom_vip_get_page_by_path` function ~and bumps the minimum WordPress version to 6.1~. 

~The alternative to a version bump would be to leave the function and wrap it in a `version_compare` statement. However, as 6.1 is nearly 2 years old, and 5 major versions ago a version bump seems logical.~ 

~Happy to revisit this PR though if we decide we don't want a WordPress version bump.~

Fixes #134

Commit:
Remove wpcom_vip_get_page_by_path

Removes the now deprecated `wpcom_vip_get_page_by_path` function and bumps the minimum WordPress vesrion to 6.1.
The alternative is to wrap the function in a `version_compare` statement but as 6.1 is nearly 2 years old, and 5 versions ago and this PR is going into a develop branch a version bump seems logical.
Fixes #134

BrookeDot commented:
Actually regarding the version bump, I see that the tests passed with WordPress 5.9, so no issue removing the VIP function, it just won't be cached on WordPress versions under 6.1 which could be an issue for some, but IMO not a big enough one to add a conditional which would only be useful if a WPVIP was on an outdated WordPress version which is against our policy, or if the `wpcom_vip_get_page_by_path` is in use on older WordPress versions outside of our platform. 

In both cases, there is very little to no risk, removing this function, and adding the version bump.

GaryJones commented:
I don't think should make this - see https://github.com/Automattic/WPCOM-Legacy-Redirector/issues/134#issuecomment-2355233829 because the `get_page_by_path()` function never changed.

GaryJones commented:
Re-opening, as the WP function did have caching added, but it was in WP 4.6.0, so it's right that we don't want plugins to be using the VIP replacement function.

GaryJones commented on wpcom-legacy-redirector.php:
```diff
@@ -5,6 +5,7 @@
  * Description: Simple plugin for handling legacy redirects in a scalable manner.
  * Version: 1.4.0-alpha
  * Requires PHP: 7.4
+ * Requires at least: 6.1
```
Let's avoid the min supported WP version bump here - since the WP core `get_page_by_path()` function had caching added in WP 4.6.0, and the plugin development branch already requires WP 5.9, we don't need to make the change (also, this was never a complete version bump anyway, since other instances would have need to have been updated, such as docs and GitHub workflows).

```suggestion
```

GaryJones committed:
Remove min WP version increase

Not needed as the core function had caching added in WP 4.6.0, and this plugin currently needs WP 5.9.
